File: lesson_secure_coding/py/broken_access_control_problem.py
import http.server
import socketserver


class MyHandler(http.server.SimpleHTTPRequestHandler):
    def do_GET(self):
        if self.path == "/admin":
            self.send_response(200)
            self.end_headers()
            self.wfile.write(b"Welcome to the admin page!")
        else:
            self.send_response(404)
            self.end_headers()
            self.wfile.write(b"Page not found.")


PORT = 8000
with socketserver.TCPServer(("", PORT), MyHandler) as httpd:
    print(f"Serving on port {PORT}")
    httpd.serve_forever()

File: lesson_secure_coding/node/broken_access_control_problem.js
const http = require('http')

const server = http.createServer((req, res) => {
    if (req.url === '/admin') {
        res.writeHead(200, { 'Content-Type': 'text/plain' });
        res.end('Welcome to the admin page!');
    } else {
        res.writeHead(404, { 'Content-Type': 'text/plain' });
        res.end('Page not found.');
    }
});

server.listen(8000, () => {
    console.log('Serving on port 8000');
});
